### PR TITLE
Switch tests to bats

### PR DESCRIPTION
**Pull request by erkannt**
As discussed in #16 this PR precedes a PR to implement the libero/ADR#9 on image tagging.

- replace existing test of `push-image.sh` using bats
- add makefile
- use makefile to simplify `ci.yml`

Currently the bats version is pinned in the make `install-deps` target. If there is a better way, please let me know. I felt that a `package.json` was overkill for a single executable with no npm dependencies.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+test-static:
+	bash -xc "shellcheck */*.sh"
+	bash -xc "find */ -name '*.sh' | xargs -I {} test -x {}"
+
+test-image_push:
+	DOCKER_REGISTRY=localhost:5000/ \
+	DOCKER_USERNAME=foo \
+	DOCKER_PASSWORD=bar \
+	$(BATS) -r tests/
```
Comment (diversemix) [inline Makefile:25]: Think you need curly braces here ${BATS}

**Review by diversemix — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,21 +8,10 @@ on:
...
-        run: |
-          bash -xc "shellcheck */*.sh"
-          bash -xc "find */ -name '*.sh' | xargs -I {} test -x {}"
-
-      - name: Tests
-        run: |
-          bash -xc "tests/*.sh"
+      - run: make setup
+      - run: make test-static
+      - run: make test-image_push
```
Comment (giorgiosironi) [inline .github/workflows/ci.yml]: what's with the mixed separators `-` and `_`? Some convention I don't know of?

File: README.md [hunk 3]
```diff
@@ -31,6 +31,21 @@ git submodule update --remote
 
 and commit the result.
 
+
+Developing
+----------
+
+If you add a shellscript consider using [bats](https://github.com/bats-core/bats-core) to add tests for it.
```
Comment (giorgiosironi) [inline README.md:38]: I think we can keep the decision of which testing framework adoption for shell script local to this project, and use it elsewhere if it proves very successful. Which means this is easy to approve.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+BATS := "node_modules/bats/bin/bats"
+
+setup: install-deps launch_registry
+
+install-deps:
+	npm install bats@1.x
```
Comment (giorgiosironi) [inline Makefile:6]: installation via NPM is weird for a bash tool, but it's suggested by their README, so who am I to judge

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	docker pull busybox
+
+launch_registry:
```
Comment (giorgiosironi) [inline Makefile]: also mixing `-` and `_`, any hidden meaning?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (giorgiosironi) [inline .github/workflows/ci.yml]: we can add the `teardown` here, which is not necessary in Github Actions but helps testing it for regressions?

File: tests/github-push-image.bats [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+@test "valid master tagging" {
+    export GITHUB_REF=refs/heads/master
+    export IMAGE_TAG=12345678
+    run docker tag busybox libero/my-dummy-project:12345678
+    run github/push-image.sh my-dummy-project
+    [ "$status" -eq 0 ]
+    run docker pull ${DOCKER_REGISTRY}liberoadmin/my-dummy-project:12345678
+    [ "$status" -eq 0 ]
+    run docker pull ${DOCKER_REGISTRY}liberoadmin/my-dummy-project:latest
+    [ "$status" -eq 0 ]
```
Comment (giorgiosironi) [inline tests/github-push-image.bats:15]: seems pretty readable and understandable if you know Bash

File: Makefile (hunk 4 above)
Comment (thewilkybarkid) [reply Makefile:6]: While we use JS elsewhere, I have a preference for using the Docker image.

Comment (erkannt) [reply Makefile:6]: I agree. For me a single `npm intall` feels odd but cleaner than dealing with nested docker or curl && ./install.sh

File: Makefile (hunk 5 above)
Comment (erkannt) [reply Makefile]: Trying to be clever. `_` being a continuation and `-` a delimiter. But consistency is better. Changed accordingly.

File: Makefile (hunk 4 above)
Comment (giorgiosironi) [reply Makefile:6]: There was a constraint on not doing nested docker, because the mounting of folders and ports would make it pretty complicated.

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**